eo-runtime/src/main/java/EOorg/EOeolang/EOseq.java

```diff
@@ -58,25 +59,33 @@ public EOseq(final Phi sigma) {
     @Override
     public Phi lambda() {
         final Phi steps = this.take("steps");
-        final Long length = new Dataized(
-            steps.take("length")
-        ).take(Long.class);
-        for (long idx = 0; idx < length - 1; ++idx) {
-            new Dataized(
-                new PhWith(
-                    steps.take("at").copy(),
-                    0, new Data.ToPhi(idx)
-                )
-            ).take();
+        final Phi[] items = toArray(steps);
+        for (int ind = 0; ind < items.length - 1; ++ind) {
+            new Dataized(items[ind]).take();
         }
         final Phi ret;
-        if (length > 0) {
+        if (items.length > 0) {
             final Phi last = steps.take("at").copy();
-            last.put(0, new Data.ToPhi(length - 1));
+            last.put(0, new Data.ToPhi((long) items.length - 1L));
             ret = last;
         } else {
             ret = new Data.ToPhi(false);
         }
         return ret;
     }
+
+    private static Phi[] toArray(final Phi args) {
+        final int length = Math.toIntExact(
+            new Dataized(
+                args.take("length")
+            ).take(Long.class)
+        );
+        final Phi[] res = new Phi[length];
+        Phi external = args;
+        for (int ind = length - 1; ind >= 0; --ind) {
+            res[ind] = new PhMethod(external, "tail");
```

Comment:
@levBagryansky why are you using `PhMethod` here? Why don't just use `external.take('tail')`